upup/pkg/fi/cloudup/apply_cluster.go

```diff
@@ -1422,6 +1424,43 @@ func (n *nodeUpConfigBuilder) BuildConfig(ig *kops.InstanceGroup, apiserverAddit
 		}
 	}
 
+	if apiModel.UseJWTForBootstrap(cluster) {
+		if !ig.HasAPIServer() {
+			keyset := keysets["bootstrap-ca"]
+			if keyset == nil {
+				return nil, nil, fmt.Errorf("key bootstrap-ca not found")
+			}
+
+			if keyset.Primary != nil && keyset.Primary.PrivateKey != nil {
```

Comment:
If either of these are nil, it should error out, not silently omit the token.

Reply:
@johngmyers if I return error and execute only `kops create cluster foobar.k8s.local` its returning error in that case. It is not executing the command successfully because its returning error from row 1435 (because private key does not exists it cannot convert it as a bytes)

Reply:
I do understand that it should return error if I have `--yes` flag, but how to avoid returning error without yes?

Reply:
I think you might need to check if the target is `DryRunTarget`, returning error only if it isn't. I'll do some research tomorrow.

Reply:
looks like that is not passed to this function either